pkg/monitortestlibrary/pathologicaleventlibrary/duplicated_event_patterns.go

```diff
@@ -2,419 +2,569 @@ package pathologicaleventlibrary
-
-	// Separated out in testNodeHasNoDiskPressure
-	regexp.MustCompile(NodeHasNoDiskPressureRegExpStr),
-
-	// Separated out in testNodeHasSufficientMemory
-	regexp.MustCompile(NodeHasSufficientMemoryRegExpStr),
-
-	// Separated out in testNodeHasSufficientPID
-	regexp.MustCompile(NodeHasSufficientPIDRegExpStr),
-
-	// Separated out in testMarketplaceStartupProbeFailure
-	regexp.MustCompile(MarketplaceStartupProbeFailureRegExpStr),
-
-	// stale condition challenge reset
-	regexp.MustCompile(`message changed from "\x{FEFF}`),
-}
-
-// AllowedUpgradeRepeatedEventPatterns are patterns of events that we should only allow during upgrades, not during normal execution.
-var AllowedUpgradeRepeatedEventPatterns = []*regexp.Regexp{
-	// Operators that use library-go can report about multiple versions during upgrades.
-	regexp.MustCompile(`ns/openshift-etcd-operator deployment/etcd-operator - reason/MultipleVersions multiple versions found, probably in transition: .*`),
-	regexp.MustCompile(`ns/openshift-kube-apiserver-operator deployment/kube-apiserver-operator - reason/MultipleVersions multiple versions found, probably in transition: .*`),
-	regexp.MustCompile(`ns/openshift-kube-controller-manager-operator deployment/kube-controller-manager-operator - reason/MultipleVersions multiple versions found, probably in transition: .*`),
-	regexp.MustCompile(`ns/openshift-kube-scheduler-operator deployment/openshift-kube-scheduler-operator - reason/MultipleVersions multiple versions found, probably in transition: .*`),
-
-	// etcd-quorum-guard can fail during upgrades.
-	regexp.MustCompile(`ns/openshift-etcd pod/etcd-quorum-guard-[a-z0-9-]+ node/[a-z0-9.-]+ - reason/Unhealthy Readiness probe failed: `),
-	// etcd can have unhealthy members during an upgrade
-	regexp.MustCompile(`ns/openshift-etcd-operator deployment/etcd-operator - reason/UnhealthyEtcdMember unhealthy members: .*`),
-	// etcd-operator began to version etcd-endpoints configmap in 4.10 as part of static-pod-resource. During upgrade existing revisions will not contain the resource.
-	// The condition reconciles with the next revision which the result of the upgrade. TODO(hexfusion) remove in 4.11
-	regexp.MustCompile(`ns/openshift-etcd-operator deployment/etcd-operator - reason/RequiredInstallerResourcesMissing configmaps: etcd-endpoints-[0-9]+`),
-	// There is a separate test to catch this specific case
-	regexp.MustCompile(RequiredResourcesMissingRegEx),
-
-	// Separated out in testMarketplaceStartupProbeFailure
-	regexp.MustCompile(MarketplaceStartupProbeFailureRegExpStr),
-}
-
-type KnownProblem struct {
-	Regexp *regexp.Regexp
-	BZ     string
-
-	// Platform limits the exception to a specific OpenShift platform.
-	Platform *v1.PlatformType
-
-	// Topology limits the exception to a specific topology (e.g. single replica)
-	Topology *v1.TopologyMode
-
-	// TestSuite limits the exception to a specific test suite (e.g. openshift/builds)
-	TestSuite *string
-}
-
-var KnownEventsBugs = []KnownProblem{
```

Comment:
I checked these as I went, virtually all were years old and unresolved, some became permanent exceptions, others were no longer occurring and were removed.